### ISSUE TITLE
[Database] Transaction Failure removes database path

### ISSUE THREAD

**Issue by dnuriel** [invertase/react-native-firebase#1503]
### Issue
```
let Ref = this.xxx.child('xxx').child(xxx).child(xxx);
        Ref.transaction(function(data){
            if (data){
                data.lastReadTime = Date.now();
                if (newMessageCount)
                    data.messagesReadCount = newMessageCount;
                else
                    data.messagesReadCount++;
                return existingUserChatData;
            }
        })
```
when receiving an error from the following code snippet the path gets removed from the database.
Can be replicated by debugging the transaction and causing a database timeout.

### Environment

1. Application Target Platform:

Both

2. Development Operating System:

macOs

3. Build Tools:

Android 26, Xcode 11

4. `React Native` version:

0.51.0

5. `React Native Firebase` Version:

4.1.0

6. `Firebase` Module:

real time database

7. Are you using `typescript`?

No

---

**Comment by dnuriel** [comment]
Ok - So I thought this issue is on both platforms but I might have been wrong can't seem to replicate it on iOS.

A temp fix - once the exception happens to use set to reinsert the data.

promise.catch(function (error) {
            //When a transaction error occurs the whole item is deleted.
            console.log(error);
            if (error.code === "DATABASE/INTERNAL-TIMEOUT"){
                Ref.set({
                    lastReadTime: Date.now(),
                    messagesReadCount: newMessageCount || 0
                });
            }
        });

**Comment by dnuriel** [comment]
OK - So an interrupt occurs 
mutableData.setValue(transactionHandler.value);
gets call on Android with transactionHandler.value = null
which removes the item from the firebase path

**Comment by Salakar** [comment]
Thanks for flagging, will investigate and get a patch out for this shortly.

---
Loving `react-native-firebase` and the support we provide? Please consider supporting us with any of the below:

  - 👉  Back financially via [Open Collective](https://opencollective.com/react-native-firebase/donate)  
  - 👉  Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter 
  - 👉  Star this repo on GitHub ⭐️

**Comment by Salakar** [comment]
Thanks for raising this, have just pushed a fix that will land in v5.1.0

---
Loving `react-native-firebase` and the support we provide? Please consider supporting us with any of the below:

  - 👉  Back financially via [Open Collective](https://opencollective.com/react-native-firebase/donate)  
  - 👉  Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter 
  - 👉  Star this repo on GitHub ⭐️
